Metalama.Framework.Engine/Linking/LinkerInjectionStep.Rewriter.cs

```diff
@@ -504,35 +536,108 @@ void AddInjectionsOnPosition( InsertPosition position )
+
+                case OperatorDeclarationSyntax { ExpressionBody: { } expressionBody } @operator:
+                    return
+                        @operator.PartialUpdate(
+                            expressionBody: null,
+                            semicolonToken: default(SyntaxToken),
+                            body: ReplaceExpression( entryStatements, expressionBody.Expression, false ) );
+
+                case PropertyDeclarationSyntax { ExpressionBody: { } expressionBody } property:
+                    Invariant.Assert( contextDeclaration is IMethod { MethodKind: Code.MethodKind.PropertyGet } );
+
+                    return
+                        property.PartialUpdate(
+                            expressionBody: null,
+                            semicolonToken: default(SyntaxToken),
+                            accessorList: AccessorList(
+                                SingletonList(
+                                    AccessorDeclaration(
+                                        SyntaxKind.GetAccessorDeclaration,
+                                        ReplaceExpression( entryStatements, expressionBody.Expression, false ) ) ) ) );
+
+                case PropertyDeclarationSyntax { AccessorList: { } accessorList } property:
+                    Invariant.Assert( contextDeclaration is IMethod { MethodKind: Code.MethodKind.PropertyGet or Code.MethodKind.PropertySet } );
+
+                    return
+                        property.PartialUpdate(
+                            accessorList: accessorList.PartialUpdate(
+                                accessors:
+                                List(
+                                    accessorList.Accessors.SelectAsArray(
+                                        a =>
+                                            IsMatchingAccessor( a, contextDeclaration )
+                                                ? a switch
+                                                {
+                                                    { Body: { } body } => a.PartialUpdate( body: ReplaceBlock( entryStatements, body ) ),
+                                                    { ExpressionBody: { } expressionBody } =>
+                                                        a.PartialUpdate(
+                                                            expressionBody: null,
+                                                            semicolonToken: default(SyntaxToken),
+                                                            body: ReplaceExpression(
+                                                                entryStatements,
+                                                                expressionBody.Expression,
+                                                                a.Kind() is not SyntaxKind.GetAccessorDeclaration ) ),
+                                                    _ => throw new AssertionFailedException( $"Not supported: {a.Kind()}" ),
```

Comment:
I take it expanding of auto-properties is handled elsewhere, so this code does not need to consider them?

Reply:
No, statements are injected into bodies in the first rewriting (injection phase), at that point, all auto-properties are still auto-properties. Deconstruction is handled in the final rewriting (linking phase).